network/src/main/scala/com/linkedin/norbert/network/partitioned/loadbalancer/gcaware/GcAwareClusteredLoadBalancerFactory.scala

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2009-2015 LinkedIn, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.linkedin.norbert.network.partitioned.loadbalancer.gcaware
+
+import com.linkedin.norbert.cluster.Node
+import com.linkedin.norbert.logging.Logging
+import com.linkedin.norbert.network.common.Endpoint
+import com.linkedin.norbert.network.partitioned.loadbalancer.{DefaultClusteredLoadBalancer, DefaultClusteredLoadBalancerFactory, PartitionedLoadBalancer}
+import com.linkedin.norbert.norbertutils.SystemClockComponent
+
+/**
+ * Extends DefaultClusteredLoadBalancer to add GC-awareness.
+ * Nodes are additionally filtered to exclude those that are currently garbage collecting.
+ *
+ * @param cycleTime: The time period in which each node in the colo undergoes garbage collection exactly once
```

Comment:
Every machine in the data center will GC once in 'cycleTime', each cluster will GC in one 'slot' of duration 'slotTime'.